### PR TITLE
Xcal launcher

### PR DESCRIPTION
**Pull request by sprout42**
Modified how command line arguments are processed to support the ability to supply an additional file (such as an XCAL, IHEX, ELF, etc) that are processed with the normal vivisect workspace mechanisms. The memory segments from a file processed this way are merged with the existing binary that is loaded through the usual flash image mechanism.

### CODE REVIEW

**Review by ma77carpenter — COMMENTED** [review]
reading through

File: cm2350/__init__.py
Comment (kuri05) [inline cm2350/__init__.py:194]: should these magic numbers be constants like 
```
MSR_MACHINECHECK
HID0_TIMEBASE
```
 or something?

File: cm2350/e200z7.py
Comment (kuri05) [inline cm2350/e200z7.py:139]: Where is `self.vw` coming from?  You removed `vw` from `__init__()`.  I assume this is requiring that the subclass populate that before initializing this?  That's at least a documentation requirement.

File: cm2350/e200z7.py
Comment (kuri05) [inline cm2350/e200z7.py:791]: lol, woops.  nice catch

File: cm2350/e200z7.py
Comment (kuri05) [reply cm2350/e200z7.py:119]: I really don't like that, actually.  I prefer handing in the `vw`.  It makes it cleaner and able to be used standalone.

File: cm2350/__init__.py
Comment (sprout42) [reply cm2350/__init__.py:194]: yes they should be

File: cm2350/e200z7.py
Comment (sprout42) [reply cm2350/e200z7.py:119]: ok, I can revisit this

File: cm2350/e200z7.py
Comment (sprout42) [reply cm2350/e200z7.py:145]: ideally, yes. I haven't come up with a good way to merge the vivisect workspaces with the MMIO capable memory regions in the emulator yet. This set of changes was intended to try and start taking more advantage of the vivisect capabilities for parsing binaries. the challenge is how best to merge those with things like the MPC5674F's FLASH peripheral. since the memory maps in the workspace probably won't be exactly the same size as the flash region.

That's just an example. I'm more than happy to talk about it. this particular workaround is intended to be just a workaround for now until we figure out how best to handle this.

File: cm2350/e200z7.py
Comment (sprout42) [reply cm2350/e200z7.py:139]: As commented below I'll change this so it's no longer a required class item.

File: cm2350/e200z7.py
Comment (ma77carpenter) [reply cm2350/e200z7.py:518]: is there a PR coming for these? :)

File: cm2350/mpc5674.py
Comment (ma77carpenter) [inline cm2350/mpc5674.py:811]: is this redundant?  
```
        ram_config = self.get_project_config('project.MPC5674.SRAM')
```

should `project` be limited here?

File: cm2350/peripherals/dspi.py
Comment (ma77carpenter) [inline cm2350/peripherals/dspi.py:10]: as long as you understand that EMULOG is a vomitous level.  It's created for Emulation pass, specifically for "isProbablyCode()" which makes lots of ugly messages.  This is *more* DEBUG than DEBUG.

Comment (ma77carpenter) [reply cm2350/peripherals/dspi.py:10]: The Viv maintainers are discussing merging this with MIRE or SHITE :)

File: cm2350/peripherals/flash.py
Comment (ma77carpenter) [inline cm2350/peripherals/flash.py:989]: ummm...  So, does this all need to change to Module?  (there are 50 mentions of Peripheral in this PR)

**Review by ma77carpenter — COMMENTED** [review]
close :)  I'll take another look after you merge/fix from the recently merged PR

File: cm2350/peripherals/flash.py
Comment (sprout42) [reply cm2350/peripherals/flash.py:989]: oh! ... I thought I changed it, lemme check

File: cm2350/peripherals/dspi.py
Comment (sprout42) [reply cm2350/peripherals/dspi.py:10]: yeah the intention was to not have these logs be part of standard DEBUG, but they can still be useful. I can put them into MIRE or SHITE if it's preferred?

File: cm2350/mpc5674.py
Comment (sprout42) [reply cm2350/mpc5674.py:811]: limited in what way?

File: cm2350/e200z7.py
Comment (sprout42) [reply cm2350/e200z7.py:518]: ah! thanks for the reminder, those are already in PPC SPRs now from an earlier PR, I'll fix this.

File: cm2350/peripherals/flash.py
Comment (sprout42) [reply cm2350/peripherals/flash.py]: oh whoops, yes eventually but not until I rebase this PR off of main.

File: cm2350/__init__.py
Comment (sprout42) [reply cm2350/__init__.py]: fixing

**Review by ma77carpenter — APPROVED** [review]
Great work, @sprout42 !

**Pull request merged**
